### PR TITLE
Allow deployments to be sorted in the api

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Add DeploymentSort to `/deployments/filter` to allow sorting deployments in the same way that flows are currently sorted.

### CODE REVIEW

File: src/prefect/orion/api/deployments.py [hunk 1]
```diff
@@ -136,6 +136,7 @@ async def read_deployments(
     flow_runs: schemas.filters.FlowRunFilter = None,
     task_runs: schemas.filters.TaskRunFilter = None,
     deployments: schemas.filters.DeploymentFilter = None,
+    sort: schemas.sorting.DeploymentSort = Body(schemas.sorting.DeploymentSort.NAME_ASC),
```
Comment (madkinsz) [inline src/prefect/orion/api/deployments.py:139]: Does sorting by default cost us query performance?

**Review by madkinsz — COMMENTED** [review]
Nice! Can you add support in `OrionClient.read_deployments` too?

File: src/prefect/orion/api/deployments.py (hunk 1 above)
Comment (madkinsz) [reply src/prefect/orion/api/deployments.py:139]: cc @chrisguidry ?

Comment (pleek91) [reply src/prefect/orion/api/deployments.py:139]: I do not know. I just tried to follow what flows is doing currently.

**Comment by pleek91** [comment]
> Nice! Can you add support in `OrionClient.read_deployments` too?

Added 👍

File: src/prefect/orion/api/deployments.py (hunk 1 above)
Comment (madkinsz) [reply src/prefect/orion/api/deployments.py:139]: 👍 We'll want to check for indices on these columns before merging as well.

Comment (zangell44) [reply src/prefect/orion/api/deployments.py:139]: Sorting by default should be fine performance-wise. We should add an index on `deployment.created` though. Happy to commit to this branch if you'd like

Comment (pleek91) [reply src/prefect/orion/api/deployments.py:139]: Please do! Thanks @zangell44

Comment (zangell44) [reply src/prefect/orion/api/deployments.py:139]: pushed!

**Review by zangell44 — APPROVED** [review]
LGTM, will get this deployed to Cloud on Monday

**Pull request merged**